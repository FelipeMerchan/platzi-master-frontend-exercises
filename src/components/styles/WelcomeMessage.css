.WelcomeMessage {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  width: 30%;
  font-family: var(--font-3);
}

.WelcomeMessage {
  margin-right: 5%;
}

.WelcomeMessage___title {
  margin-bottom: var(--space);
  padding-bottom: var(--space);
  border-bottom: 1px solid #dfdfdf;
  color: #48504d;
}

.WelcomeMessage__text {
    margin-bottom: var(--space-2);
    line-height: 1.4;
    color: #a6a7a2;
}

.WelcomeMessage__text a {
  color: var(--actionColor);
}