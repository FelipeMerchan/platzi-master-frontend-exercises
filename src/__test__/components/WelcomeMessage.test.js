import React from 'react'
import { shallow } from 'enzyme'
import { create } from 'react-test-renderer'
import '../../setupTest'
import WelcomeMessage from '../../components/WelcomeMessage'

describe('Probar el componente <WelcomeMessage />', () => {
  const welcomeMessage = shallow(<WelcomeMessage />)
  test('Render del componente WelcomeMessage', () => {
    expect(welcomeMessage.length).toEqual(1)
  })
  test('Render del título', () => {
    expect(welcomeMessage.find('.WelcomeMessage___title').text()).toEqual('¡Bienvenido!')
  })
})

describe('WelcomeMessage Snapshot', () => {
  test('Comprobar la UI del componente WelcomeMessage', () => {
    const welcomeMessage = create(<WelcomeMessage />)
    expect(welcomeMessage.toJSON()).toMatchSnapshot()
  })
})