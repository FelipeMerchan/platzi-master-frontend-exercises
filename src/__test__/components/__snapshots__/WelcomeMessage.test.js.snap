// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeMessage Snapshot Comprobar la UI del componente WelcomeMessage 1`] = `
<section
  className="WelcomeMessage"
>
  <h2
    className="WelcomeMessage___title"
  >
    ¡Bienvenido!
  </h2>
  <p
    className="WelcomeMessage__text"
  >
    Lorem ipsum dolor sit amet, consectetuer adipiscing elit. 
    <a
      href="/"
      target="_blank"
    >
      Aenean massa.
    </a>
     Ut wisi enim ad minim veniam, quis nostrud exerci tation ullamcorper suscipit lobortis nisl 
    <a
      href="/"
      target="_blank"
    >
      Nuliam dictum
    </a>
     ex ea commodo consequat.
  </p>
  <p
    className="WelcomeMessage__text"
  >
    Duis autem vel eum iriure dolor in hendrerit in vulputate velit esse molestie consequat, vel illum dolore eu feugiat nulla facilisis at vero eros et accumsan et iusto odio dignissim qui blandit praesent luptatum zzril delenit augue duis dolore te feugait nulla facilisi.
  </p>
  <div
    className="mt-2"
  >
    <a
      className="button"
      href="/"
      target="_blank"
    >
      Leer más
    </a>
  </div>
</section>
`;
